client/rest-high-level/src/test/java/org/opensearch/client/indices/GetIndexTemplatesResponseTests.java

```diff
@@ -99,8 +102,12 @@ public void testParsingFromOpenSearchResponse() throws IOException {
                 esIMD.settings(randomIndexSettings());
                 esIMD.putMapping("_doc", new CompressedXContent(BytesReference.bytes(randomMapping("_doc", xContentType))));
                 int numAliases = randomIntBetween(0, 8);
+                Set<String> uniqueAliases = new HashSet<>();
                 for (int j = 0; j < numAliases; j++) {
-                    esIMD.putAlias(randomAliasMetadata(String.format(Locale.ROOT, "%02d ", j) + randomAlphaOfLength(4)));
+                    uniqueAliases.add(String.format(Locale.ROOT, "%02d ", j) + randomAlphaOfLength(4));
+                }
+                for (String uniqueAlias : uniqueAliases) {
+                    esIMD.putAlias(randomAliasMetadata(uniqueAlias));
```

Comment:
There doesn't seem to be a check that uniqueAliases actually has `numAliases` number of elements in it. When we add a duplicate element, the add method on line 107 should return false - a while loop check till it returns true will ensure we have the correct number of elements in uniqueAliases.